h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -0,0 +1,144 @@
+package hex.genmodel.algos.gam;
+
+import hex.genmodel.GenModel;
+import hex.genmodel.MojoModel;
+import hex.genmodel.easy.RowData;
+import hex.genmodel.utils.ArrayUtils;
+
+public abstract class GamMojoModelBase extends MojoModel {
+  String _link;
+  boolean _useAllFactorLevels;
+  int _cats;
+  int[] _catNAFills;
+  int[] _catOffsets;
+  int _nums;
+  int _numsCenter;
+  double[] _numNAFills;
+  double[] _numNAFillsCenter;
+  boolean _meanImputation;
+  double[] _beta;
+  double[] _beta_no_center;
+  double[] _beta_center;
+  double[][] _beta_multinomial;
+  double[][] _beta_multinomial_no_center; // coefficients not centered for multinomial/ordinal
+  double[][] _beta_multinomial_center; // coefficients not centered for multinomial/ordinal
+  String _family;
+  String[] _gam_columns;
+  int _num_gam_columns;
+  int[] _bs;
+  int[] _num_knots;
+  double[][] _knots;
+  double[][][] _binvD;
+  double[][][] _zTranspose;
+  String[][] _gamColNames;  // expanded gam column names
+  String[][] _gamColNamesCenter;
+  String[] _names_no_centering; // column names of features with no centering
+  int _totFeatureSize; // number of predictors plus gam columns no centered
+  int _betaSizePerClass;
+  int _betaCenterSizePerClass;
+  double _tweedieLinkPower;
+  double[][] _basisVals; // store basis values array for each gam column
+  double[][] _hj;   // difference between knot values
+  int _numExpandedGamCols; // number of expanded gam columns
+  int _lastClass;
+  
+  public int get_totFeatureSize() { return _totFeatureSize;}
```

Comment:
please respect java naming convention when naming methods and vars, same below

Reply:
Sure.